File: longest-consecutive-sequence/longest-consecutive-sequence.py
from sortedcontainers import SortedSet
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
 
        
        nums = SortedSet(nums)
        res  = 0 
        cnt = 0 
        for num in nums:
            if num-1 in nums:
                cnt += 1
            else:
                res = max(res,cnt)
                cnt = 1
        res = max(res,cnt)
        return res

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> st ;
        for(int num : nums){
            st.insert(num);
        }
        int maxi = 0 ; 
        for(int num: nums){
            if(!st.count(num-1)){
                int curr = num ; 
                int cnt = 1 ; 
                while(st.count(curr+1)){
                    curr += 1;
                    cnt += 1;
                }
                maxi = max(maxi , cnt);
            }
        }
        return maxi ;
    }
};
